src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2768,7 +2768,26 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			title: nls.localize('TaskService.fetchingBuildTasks', 'Fetching build tasks...')
 		};
 		let promise = (async () => {
-			const buildTasks = await this._findWorkspaceTasksInGroup(TaskGroup.Build, false);
+
+			let buildTasks: (Task | ConfiguringTask)[] = [];
+
+			// First check for globs before checking for default build tasks
+			const uri = EditorResourceAccessor.getOriginalUri(this.editorService.activeEditor);
```

Comment:
Currently the URI is the absoulte path of the file, i think this should only be from the workspace root but have no idea how to limit this, I may need help here

Reply:
You can use the `IWorkspaceContextService` to get the workspace folder for a file:

https://github.com/jasonwilliams/vscode/blob/a0a448f344bc8c068deb1202708dfc1e7c17ef89/src/vs/platform/workspace/common/workspace.ts#L64-L65